_search-plugins/search-pipelines/index.md

```diff
@@ -0,0 +1,387 @@
+
+![Search processor diagram]({{site.url}}{{site.baseurl}}/images/search-pipelines.png)
+
+Both request and response processing for the pipeline are performed on the coordinator node so there is no shard-level processing.
+{: .note}
+
+## Search request processors
+
+OpenSearch supports the following search request processors:
+
+- [`script`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/script-processor/): Adds a script that is run on newly indexed documents.
+- [`filter_query`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/filter-query-processor/): Adds a filtering query that is used to filter requests.
+
+## Search response processors
+
+OpenSearch supports the following search response processors:
+
+- [`rename_field`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/rename-field-processor/): Renames an existing field.
+
+## Viewing available processor types
+
+You can use the Nodes Search Pipelines API to view the available processor types:
+
+```json
+GET /_nodes/search_pipelines
+```
+{% include copy-curl.html %}
+
+The response contains the `search_pipelines` object that lists the possible request and response processors:
+
+<details open markdown="block">
+  <summary>
+    Response
+  </summary>
+  {: .text-delta}
+
+```json
+{
+  "_nodes" : {
+    "total" : 1,
+    "successful" : 1,
+    "failed" : 0
+  },
+  "cluster_name" : "runTask",
+  "nodes" : {
+    "36FHvCwHT6Srbm2ZniEPhA" : {
+      "name" : "runTask-0",
+      "transport_address" : "127.0.0.1:9300",
+      "host" : "127.0.0.1",
+      "ip" : "127.0.0.1",
+      "version" : "3.0.0",
+      "build_type" : "tar",
+      "build_hash" : "unknown",
+      "roles" : [
+        "cluster_manager",
+        "data",
+        "ingest",
+        "remote_cluster_client"
+      ],
+      "attributes" : {
+        "testattr" : "test",
+        "shard_indexing_pressure_enabled" : "true"
+      },
+      "search_pipelines" : {
+        "request_processors" : [
+          {
+            "type" : "filter_query"
+          },
+          {
+            "type" : "script"
+          }
+        ],
+        "response_processors" : [
+          {
+            "type" : "rename_field"
+          }
+        ]
+      }
+    }
+  }
+}
+```
+</details>
+
+In addition to the processors provided by OpenSearch, additional processors may be provided by plugins.
+{: .note}
+
+## Creating a search pipeline
+
+Search pipelines are stored in the cluster state. To create a search pipeline, you must configure an ordered list of processors in your OpenSearch cluster. You can have more than one processor of the same type in the pipeline. Each processor has a `tag` identifier that distinguishes it from the others. Tagging a specific processor can be helpful for error messages, especially if you add multiple processors of the same type.
```

Comment:
The meaning of "for" in "for error messages" is a little ambiguous. Should it just be "in", or is there additional clarification we can add (in what way is it helpful)?